hal/network/lwip/wiznet/wiznetif_config.h

```diff
@@ -0,0 +1,131 @@
+/*
+ * Copyright (c) 2023 Particle Industries, Inc.  All rights reserved.
+ *
+ * This library is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation, either
+ * version 3 of the License, or (at your option) any later version.
+ *
+ * This library is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public
+ * License along with this library; if not, see <http://www.gnu.org/licenses/>.
+ */
+
+#pragma once
+
+// STATIC_ASSERT already defined from static_assert.h as PARTICLE_STATIC_ASSERT
+#ifdef STATIC_ASSERT
+#undef STATIC_ASSERT
+#endif
+// app_util.h unconditionally defines STATIC_ASSERT
+#include "static_recursive_mutex.h"
+#include "platforms.h"
+
+#ifdef __cplusplus
+
+namespace particle {
+
+namespace net {
+
+#if PLATFORM_ID == PLATFORM_ARGON || PLATFORM_ID == PLATFORM_BORON || PLATFORM_ID == PLATFORM_P2 || PLATFORM_ID == PLATFORM_TRACKERM
+#define WIZNETIF_CS_PIN_DEFAULT    (D5)
```

Comment:
Might be a good idea to perhaps move these into `hal_platform_config.h`?

Reply:
I ended up moving them after our discussion this morning.